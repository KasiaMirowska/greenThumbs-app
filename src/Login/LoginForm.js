import React from 'react';
import TokenService from '../Services/token-service';
import AuthCalls from '../Services/Auth_Service';
import GreenContext from '../Context';

export default class LoginForm extends React.Component {

   static contextType = GreenContext;
   state = {
       error: null,
   }

    handleForm =(e)=> {
        e.preventDefault();
        const {username, password} = e.target

        const credentials = {
            username: username.value,
            password: password.value,
        }
        
        AuthCalls.postLogin(credentials)
        .then(res => {
            username.value = '';
            password.value = '';
            TokenService.saveAuthToken(res.authToken);
            this.props.onLoginSuccess();
        })
        .catch(err => {
            console.log(err)
            this.context.setError(err)
        })
    }

    handleUserNameCleanUp = (e) => {
        e.preventDefault();
        let userName = e.target;
        if(userName.value !== null) {
            userName.value = '';
            this.context.resetError();
        } 
    }

    handlePasswordCleanUp = (e) => {
        e.preventDefault();
        let password = e.target;
        if(password.value !== null) {
           password.value = '';
           this.context.resetError();
        }  
    }
    
    render() {
        return (
            <div>
                <form className='auth-form' onSubmit={this.handleForm}>
                    <input className="form__field form__field3" type='input' id='username' placeholder='username' required onClick={this.handleUserNameCleanUp}/>
                    <br />
                    <input className="form__field form__field3" type='password' id='password' placeholder='password' required onClick={this.handlePasswordCleanUp} />
                    <button type='submit'>Submit</button>
                    <button className='login-button' type="reset" onClick={this.handleAfterErrorDisplay}>
                        Cancel
                    </button>
                </form>
            </div>
        )
    }
}