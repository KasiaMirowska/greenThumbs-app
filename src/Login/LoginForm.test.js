import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter as BR} from 'react-router-dom';
import LoginForm from './LoginForm';


describe('List component', () => {
    const props = {
        onLoginSuccess: () => {},
    }
   
    it ('renders without crashing',() => {
        const div = document.createElement('div');
        ReactDom.render(<BR><LoginForm {...props} /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
   
    it('renders UI as expected', () => {
        const item= renderer.create(<BR><LoginForm {...props} /></BR>);
        expect(item.toJSON()).toMatchSnapshot();
    })
})