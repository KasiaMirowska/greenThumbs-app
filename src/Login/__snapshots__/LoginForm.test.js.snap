// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders UI as expected 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      id="username"
      placeholder="username"
      type="input"
    />
    <input
      id="password"
      placeholder="password"
      type="password"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
